### ISSUE TITLE
Separate method for Feature Engineering

### ISSUE THREAD

**Issue by inverse-panda** [AutoViML/featurewiz#77]
Thanks for the nice library, I find quite useful, but I have some trouble using the feature engineering part.  the code below is an example how I am using the  library.

```
data = pd.DataFrame(
    {
        "Color": ["Red", "Blue", "Green", "Red", "Green", "Red", "Blue"],
        "Size": ["S", "M", "L", "XL", "M", "XL", "S"],
        "Shape": [
            "Circle",
            "Square",
            "Triangle",
            "Circle",
            "Square",
            "Circle",
            "Triangle",
        ],
        "Temperature": ["Hot", "Cold", "Hot", "Warm", "Warm", "Cold", "Hot"],
       "Target": [1, 0, 1, 1, 0, 0, 1]
    }
)
X_train = data.drop("Target", axis=1)
y_train = data["Target"]

X_test = pd.DataFrame(
    {
        "Color": ["Red", "Blue", "Green", "Red", ],
        "Size": ["S", "M", "L", "XL",],
        "Shape": [
            "Circle",
            "Square",
            "Triangle",
            "Circle",
        ],
        "Temperature": ["Hot", "Cold", "Hot", "Warm",],
    }
)

fwiz = FeatureWiz(
    corr_limit=0.7,
    feature_engg=[
        "groupby",
        "interactions",
    ],
    category_encoders=["HashingEncoder", "OneHotEncoder"],
    dask_xgboost_flag=False,
    nrows=None,
    verbose=2,
)

X_train_selected = fwiz.fit_transform(
    X=X_train,
    y=y_train,
)

X_test_selected  = fwiz.transform(X_test)

print(X_train_selected) #=> (7, 9)
print(X_test.shape)  # => (4,4)
print(X_test_selected) #=> (7, 9)
```
I get a printed message, which does not imply the `transfrom` method is returning the training set. I find the feature engineering parts, very useful. Is there a way to run feature generation on the test set (ideally only for selected features).

**Comment by AutoViML** [comment]
Let me see what I can do - do you have any idea of what the syntax should look like for columns that need feature engg?

**Comment by inverse-panda** [comment]
some tools implement the feature_engg as simple methods with this signature,
```
def feature(x: List[Any], **params) -> List[float | bool]:
    pass
```
look at [tsFresh](https://tsfresh.readthedocs.io/en/latest/text/how_to_add_custom_feature.html). using this you can serialize the feature engg, all you need for each feature are:
1. the name of the feature engg method
2. the columns names that are input to the the feature
3. the extra parameters of the method

It gives the flexibility to user to run the feature engg and selection, in the training pipeline, and use the result to only and only run the feature engg on for the selected columns in the test pipeline.

**Comment by AutoViML** [comment]
hmmm... that's very interesting. In light of featurewiz, can you run a quick scan of the feature engg methods I have and name some functions that I should start with?If you remember, I already have scikit-learn compatible transformers for time series features, categorical transformations, etc. Those transformers should left as is I assume since they can be chained together in a pipeline using the sklearn Pipeline object. I am not sure if there are others. Let's work together and see what comes out.Auto Vimal

    On Thursday, July 6, 2023 at 07:31:25 AM EDT, inverse-panda ***@***.***> wrote:  
 
 


some tools implement the feature_engg as simple methods with this signature,
def feature(x: List[Any], **params) -> List[float | bool]:
    pass

look at tsFresh. using this you can serialize the feature engg, all you need for each feature are:
   
   - the name of the feature engg method
   - the columns names that are input to the the feature
   - the extra parameters of the method

It gives the flexibility to user to run the feature engg and selection, in the training pipeline, and use the result to only and only run the feature engg on for the selected columns in the test pipeline.
